lib/ansible/modules/cloud/digital_ocean/digital_ocean_space.py

```
@@ -0,0 +1,198 @@
+  region:
+    description:
+    - Region where the space will reside.
+    default: 'nyc3'
+  canned_acl:
+    description:
+    - Canned ACL to apply to the bucket.
+    default: private
+    choices: ['private', 'public-read']
+  access_id:
+    description:
+    - Access ID used for authentication with Digital Ocean Spaces API (Default is DO_ACCESS_KEY_ID).
+    default: ''
+  secret_key:
+    description:
+    - Secret key used for authentication with Digital Ocean Spaces API (Default is DO_SECRET_ACCESS_KEY).
+    default: ''
+extends_documentation_fragment:
+    - digital_ocean.documentation
+    - aws
+notes:
+  - Two environment variables are used for authentication with the Digital Ocean
+    Spaces API.  They are DO_ACCESS_KEY_ID and DO_SECRET_ACCESS_KEY.  You can
+    create this key pair in the Digital Ocean control panel.
+'''
+
+
+EXAMPLES = r'''
+- name: create a space
+  digital_ocean_space:
+    name: cool_example_space
+    state: present
+
+- name: remove a space
+  digital_ocean_space:
+    name: cool_example_space
+    state: absent
+
+- name: use API credentials stored in ansible-vault
+  digital_ocean_space:
+    name: cool_example_space
+    state: present
+    access_id: "{{ super_secure_access_id_stored_in_vault }}"
+    secret_key: "{{ super_secure_secret_key_also_stored_in_vault }}"
+
+'''
+
+
+RETURN = r'''#'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.digital_ocean import DigitalOceanHelper
+from ansible.module_utils._text import to_native
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def space_exists(client, module):
+    space_name = module.params.get('name')
+    try:
+        client.head_bucket(Bucket=space_name)
+        return True
+    except ClientError:
+        return False
```

Is this just ignoring an error? Should we give some details about the exception? Per documentation (http://botocore.readthedocs.io/en/latest/client_upgrades.html#error-handling) we can at least print the error message **.msg**